Sources/ElasticSwift/Requests/DeleteRequest.swift

```diff
@@ -49,113 +53,57 @@ public class DeleteRequestBuilder: RequestBuilder {
         return self
     }
     
-    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> Void) -> Self {
+    public func set(completionHandler: @escaping (DeleteResponse?, Error?) -> ()) -> Self {
         self.completionHandler = completionHandler
         return self
     }
     
-    public func build() -> Request {
+    public func make() throws -> DeleteRequest {
         return DeleteRequest(withBuilder: self)
     }
-    
+
 }
 
 public class DeleteRequest: Request {
+   
+    public typealias ResponseType = DeleteResponse
     
-    let client: ESClient
+    public var completionHandler: ((DeleteResponse?, Error?) -> ())?
```

Comment:
I've made the completionHandler Optional, I think sometime the caller don't need to manage the response.
This also  avoid some force unwrapping on builder.completionHandler!